src/peergos/server/util/TimeLimited.java

```diff
@@ -0,0 +1,31 @@
+package peergos.server.util;
+
+import peergos.shared.cbor.*;
+import peergos.shared.crypto.asymmetric.*;
+import peergos.shared.crypto.hash.*;
+import peergos.shared.storage.*;
+
+import java.util.*;
+
+public class TimeLimited {
+
+    public static boolean isAllowedTime(byte[] signedTime, int durationSeconds, ContentAddressedStorage ipfs, PublicKeyHash owner) {
+        try {
+            Optional<PublicSigningKey> ownerOpt = ipfs.getSigningKey(owner).get();
+            if (! ownerOpt.isPresent())
+                throw new IllegalStateException("Couldn't retrieve owner key during getFollowRequests() call!");
+            byte[] raw = ownerOpt.get().unsignMessage(signedTime);
+            CborObject cbor = CborObject.fromByteArray(raw);
+            if (! (cbor instanceof CborObject.CborLong))
+                throw new IllegalStateException("Invalid cbor for getFollowRequests authorisation!");
+            long utcMillis = ((CborObject.CborLong) cbor).value;
+            long now = System.currentTimeMillis();
+            if (Math.abs(now - utcMillis) > durationSeconds * 1_000)
+                throw new IllegalStateException("Stale auth time, is your clock accurate?");
```

Comment:
to aid debugging: add ``utcMillis`` (formatted as a date-time) and duration to the exception message.

Reply:
I was part way through implementing this when I realised that both times are always UTC, so reporting a time in utc will probably confuse the user more given they could be in any time zone.